csv_permissions/evaluators.py

```diff
@@ -0,0 +1,76 @@
+from typing import Optional
+import warnings
+
+from csv_permissions.types import Evaluator
+from csv_permissions.types import UnresolvedEvaluator
+
+
+def make_evaluate_not_implemented(message) -> Evaluator:
+    """Creates a new evaluator that when invoked throws a NotImplementedError error
+    with a given message"""
+    def evaluate_not_implemented(user, obj=None) -> bool:
+        raise NotImplementedError(message)
+    return evaluate_not_implemented
+
+
+def resolve_validation_evaluator(details: UnresolvedEvaluator) -> Optional[Evaluator]:
+    """
+    This doesn't actually resolve to any evaluators, it just performs some validation
+    """
+    if details.model is None and not details.is_global:
+        raise ValueError("Permissions without Models must be global.")
```

Comment:
as noted above - is this a hard requirement? does it matter if you wanted to use a non-model permission as an object level permission with some arbitrary object?

Reply:
(see above; added comment, may relax restriction in future)